test/pipelines/test_pipeline_yaml.py

```diff
@@ -140,6 +140,30 @@ def test_load_yaml(tmp_path):
     assert isinstance(pipeline.get_node("reader"), MockReader)
 
 
+def test_load_yaml_null_values(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+version: ignore
+
+components:
+  - name: MyReader
+    type: FARMReader
+    params:
+      model_name_or_path: deepset/tinyroberta-squad2
+      model_version: null         
```

Comment:
I would add tests for the case of `Optional[List[Any]]` and `Optional[Dict[str, Any]]` to make sure everything works well.

Reply:
Added the tests.
For the optional dict, I only found `custom_mapping` in `ElasticsearchDocumentStore`, and I test it.
If you have better suggestions (not involving Elasticsearch), I'm all ears. :smiley: 

*Please note that the tests are testing pipeline validation with JSON master schema and not schema generation.*

Reply:
A better way would be to make a custom node that takes these three optional params, a bit like it's done here: https://github.com/deepset-ai/haystack/blob/d5e36ce6b4b61a1237a6b5c76d0571b9aef01de2/test/pipelines/test_pipeline_yaml.py#L296-L320 This one takes only one `int` params but I think you get the idea.

It would be actually best to use only mock nodes in the unit tests, so with this fix we're killing two birds with one stone :grin:

Reply:
Great suggestion!

I replaced the tests, going in this direction...